generic.c

```diff
@@ -164,3 +164,30 @@ int createFile(const char * file, const char * buff, size_t size)
 	return SUCCESS;
 }
 
+/*
+ *returns a new pointer to an array with new length
+ *@param arr , a pointer to the array, will be reallocated to have desired size or NULL if error
+ *@param new_length , the desired length of the array
```

Comment:
Is this length of the array, or size of the array? This takes in `void **` and the calls to this function cast away the type. Length implies a counted number of elements in the array, which when reallocating, we would need to know the size of each element to be able to allocate an appropriate buffer size.

I would suggest making this more clear by either taking in the *size*, forcing the caller to do the math on the size, or also take in the element size as a third argument.

Reply:
made this fucntion more like `calloc` with number of elements and size of each element arguments.